### DISCUSSION TITLE
Using settings in routing (MissingSettings when running initial migrations)

### DISCUSSION THREAD

**Discussion by pluma** [spatie/laravel-settings#285] [Q&A]
I want to be able to define certain routes at runtime and using settings seems like the obvious place to do this. But if I try to use `app(MySettings::class)` in my `web.php` of course this throws an exception when running migrations necessary to initialize those settings. Wrapping this call in a try/catch doesn't seem to have any effect.

Is there a way to check if settings have already been migrated so I can wrap this in an if-block or am I missing something obvious?

**Comment by rubenvanassche** [comment]
As far as I'm aware there isn't a solution to this since routes will always be mapped even when running migrations and thus the required settings cannot be found.

**Comment by pluma** [comment]
A workaround for my use case seems to be to check `\Illuminate\Support\Facades\Schema::hasTable('settings')` which will return `false` if the settings table doesn't exist yet (which would cause the exception I was seeing). This will probably not help with specific settings added in later migrations but it does solve the problem when running `migrate:fresh`.
